XCSoar (.CUP) and Winpilot (.DAT) file processing now works

paynterf opened the pull request:
XCSoar (.CUP) and Winpilot (.DAT) file processing now works properly.  Unfortunately I couldn't figure out how to get the left/right/top/bottom bounds values to show up in the corresponding web page windows, so the map still isn't generated automatically.

As a workaround the calculated bounds are displayed in the 'error' window, and the user can manually transcribe the values to the bounds entry fields and then select the 'select map bounds by hand' option.

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -75,27 +77,65 @@ def __parse_length(str):
+    wpnum = 0
+    for byteline in lines:
+        wpnum = wpnum + 1
+        line = byteline.decode('ISO-8859-2') #gfp 241210: added 'ISO-8859-2' decoding for correct cherrypy logging display
```
Shouldn't this be UTF-8?

paynterf replied:
I tried 'UTF-8' and AFAICT, it doesn't handle the accent marks in the Slovenia3 waypoint list.  I added the 'Slovenia3.cup' and 'Slovenia3.dat' test files for your consideration (the .DAT file was generated from the .CUP file using a separate Python script to make sure they both contained the exact same coordinates)

I have no problem using UTF-8 vs ISO-8859-2, but some of the characters in Slovenia3.cup won't be displayed properly. I don't think this will affect the bounds calculation either way.

lordfolken replied:
The files you provide contain corrupted data.
Slovenia3.cup: Lines 26, 94, 774, 776. 

A CUP file should be utf-8. https://github.com/naviter/seeyou_file_formats/blob/main/CUP_file_format.md

These files should only be included if they serve an automated test case, and then they should be under test/data/cup/filename.

lordfolken replied:
https://download.xcsoar.org/content/waypoint/country/SI-WPT-National-OpenAIP.cup if you need an example that works.

lordfolken replied:
https://www.condorsoaring.com/forums/viewtopic.php?f=29&t=22737&p=191905#p191905 seems to be the file from slovenia3 in condor.

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -75,27 +77,65 @@ def __parse_length(str):
+        # cherrypy.log('in for loop: wpnum = %s line = %s' %(wpnum, line))
+#        cherrypy.log(f'for loop row {wpnum}: {line}')
+
+        #check for blank lines or comments
```
Lots of debugging code. We don't really want that in the code base.

lordfolken reviewed (changes requested):
Congrats on figuring out the basic problem. :)

Can you please make a change that is as minimal as possible to fix the cup parsing issue and make a commit for it? No Test files or commented out debug code please, also no excessive logging which isn´t needed for operation.

paynterf commented:
Let me get my Linux laptop back out and I'll do the changes you requested ;)

paynterf commented:
Regarding the .CUP file encoding issue, I checked both the Slovenia3.cup and the Slovenia3.cup, and it appears the Slovenia3.cup file is corrupted - at least in my Landscapes/Slovenia3 folder.  I did see that Slovenia3.cup is encoded in UTF-8 as you stated, but for some reason the Slovenia2.cup file is encoded as 'ANSI'.  In any case I'll change back to UTF-8, but what to do about the corrupted lines in Slovenia3.cup?  Can you check Slovenia3.cup on your end?  Here is my line for 'Amzs Koa' (which isn't in the Slovenia2.cup file)

"Amzs Koâ”œÃ",Amzs Koì,4539.139N,01450.914E,470.0m,1,,,,

paynterf commented:
commit 2abe1a4d75f8db84fbb20e179108923fbcfac370 (HEAD -> master)
Author: Frank Paynter <frank@M6700.local>
Date:   Wed Jan 1 21:47:37 2025 -0500

    Changed decoding to UTF-8, removed logging code, removed test files

lordfolken commented:
> Regarding the .CUP file encoding issue, I checked both the Slovenia3.cup and the Slovenia3.cup, and it appears the Slovenia3.cup file is corrupted - at least in my Landscapes/Slovenia3 folder. I did see that Slovenia3.cup is encoded in UTF-8 as you stated, but for some reason the Slovenia2.cup file is encoded as 'ANSI'. In any case I'll change back to UTF-8, but what to do about the corrupted lines in Slovenia3.cup? Can you check Slovenia3.cup on your end? Here is my line for 'Amzs Koa' (which isn't in the Slovenia2.cup file)
> 
> "Amzs Koâ”œÃ",Amzs Koì,4539.139N,01450.914E,470.0m,1,,,,

I also have condor3 and i see the same issue in the Slovenia3.cup file. So its really corrupted.

lordfolken commented on Slovenia3.cup:
Please remove this cup/dat files for now. While i understand your testing intent, we don't want that in the production code base.

lordfolken commented on lib/xcsoar/mapgen/server/server.py:
```diff
@@ -12,6 +12,14 @@
 from xcsoar.mapgen.georect import GeoRect
 from xcsoar.mapgen.waypoints.parser import parse_waypoint_file
 
+cherrypy.config.update(
+    {
+        "log.screen": True,  # Log to stdout
+        "log.error_file": "error.log",  # Log errors to a file
+        "log.access_file": "access.log",  # Log access to a file
+    }
+)
+
```
Normally this runs in a container. The Filesystem is usually readonly. Please remove this configuration.

lordfolken commented on lib/xcsoar/mapgen/server/views/index.html:
```diff
@@ -57,7 +57,7 @@
       <em class="note">Draw the map bounds rectangle by clicking in the map.</em>
       <hr/>
-      <label class="left" for="left">Minimum Longitude:</label><input id="left" type="text" name="left"/><br/>
+      <label class="left" for="left">Minimum Longitude:</label><input id="left" type="text" name="left" value="43"/><br/>
```
why the default value=43?

paynterf replied:
I think that snuck in there while I was trying to figure out how to display the waypoint bounds.  I have removed it.

lordfolken commented on lib/xcsoar/mapgen/waypoints/list.py:
```diff
@@ -33,10 +34,13 @@ def extend(self, wp_list):
     def get_bounds(self, offset_distance=15.0):
         rc = GeoRect(180, -180, -90, 90)
         for wp in self.__list:
+            # cherrypy.log(f'In list.py: {wp.name}, lat: {wp.lat:.3f}, lon: {wp.lon:.3f}')
```
please remove.

lordfolken commented on lib/xcsoar/mapgen/waypoints/list.py:
```diff
@@ -33,10 +34,13 @@ def extend(self, wp_list):
             rc.bottom = min(rc.bottom, wp.lat)
 
         rc.expand(offset_distance)
+        # cherrypy.log(f'In list.py - final rc: left {rc.left:.3f}, right: {rc.right:.3f}, top: {rc.top:.3f}, bot {rc.bottom:.3f}')
+
```
please remove.

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -133,3 +135,5 @@ def parse_seeyou_waypoints(lines, bounds=None):
         waypoint_list.append(wp)
 
     return waypoint_list
+
+    return waypoint_list
```
I don't quite understand this double return statement?

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -76,26 +76,28 @@ def __parse_length(str):
-        if first:
-            first = False
-            continue
+    header = "name,code,country,lat,lon,elev,style,rwdir,rwlen,freq,desc"
```
beware that the there is a new standard since 2022, that has some additional fields:
https://github.com/naviter/seeyou_file_formats/blob/main/CUP_file_format.md

Maybe xcsoar's parser can serve as a reference:
https://github.com/XCSoar/XCSoar/blob/master/src/Waypoint/WaypointReaderSeeYou.cpp

paynterf commented:
Removed another couple of logging code lines, and added Slovenia3.cup/dat to .gtignore

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -75,27 +77,65 @@ def __parse_length(str):
+        # for field in fields:
+        #     cherrypy.log(f' field[{idx}] = {field}')
+        #     idx += 1
+
```
in general. no commented out code please. :-)

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -75,27 +77,65 @@ def __parse_length(str):
+    # wpnum = 0
+    # for byteline in lines:
+    #     wpnum = wpnum + 1
+    #     line = byteline.decode('ISO-8859-2')
```
please remove

lordfolken commented on lib/xcsoar/mapgen/server/server.py:
```diff
@@ -91,12 +114,30 @@ def index(self, **params):
+                cherrypy.log(f'in server.py: {filename} bounds: left = {desc.bounds.left:.3f}, right: {desc.bounds.right:.3f}, top: {desc.bounds.top:.3f}, bot {desc.bounds.bottom:.3f}')
+                return view.render(error=f"left: {desc.bounds.left:.3f}, right: {desc.bounds.right:.3f}, top: {desc.bounds.top:.3f}, bot {desc.bounds.bottom:.3f}")| HTMLFormFiller(data=params)
+                # return view.render(error="left")| HTMLFormFiller(data=params)
+                time.sleep(9)
```
Why sleep?

lordfolken reviewed (changes requested):
So we are getting there. :+1: 

Please remove any unessential changes. 
No unnecessary newlines, indention changes, 
Please delete Slovenia* from the code tree
also still some cherrypy.log entries which are just debug outputs

Regards, 
 - Folken

paynterf commented:
One comment line removed from server.py, Slovenia* removed.

paynterf commented:
Haven't seen anything here recently.  Do you have any other requested changes?

paynterf commented:
Just checking in here.  I see you have requested a change, but I can't see what the requested change is.  Am I missing something?

lordfolken commented on lib/xcsoar/mapgen/waypoints/seeyou_reader.py:
```diff
@@ -76,26 +76,28 @@ def __parse_length(str):
+    wpnum = 0
+    for byteline in lines:
+        wpnum = wpnum + 1
+        line = byteline.decode("UTF-8")
```
@coderabbitai this would incur silent data loss. This would be risky in an aviation context.

lordfolken replied:
@coderabbitai please create a pr for your suggestion

lordfolken commented:
@coderabbitai hey bunny, can you analyze server.py and see why the bounds would not get parsed when uploading a foobar.cup file?
